### ISSUE TITLE
When editing content in the middle of the word, the word highlight is not automatically canceled.

### ISSUE THREAD

**Issue by AGou-ops** [nyngwang/murmur.lua#1]
Or it will be automatically highlighted when editing at the end of the word, so that it looks more uniform, just like vscode.

https://user-images.githubusercontent.com/57939102/204951002-e04c0b70-955f-40ee-a05b-453a8ca44b4a.mp4

Is there any solution?

**Comment by nyngwang** [comment]
@AGou-ops Let me give you a "Wow". I have been using this plugin for about 6 months and I didn't notice this difference. Good catch!

> Or it will be automatically highlighted when editing at the end of the word [...]

I prefer this behavior more since sometimes I need to make sure that I type the same word as the existing one. Is this OK for you? Or should I make this a toggle so that each user can choose what they prefer?

**Comment by AGou-ops** [comment]
Thanks for your reply!

>  Or should I make this a toggle so that each user can choose what they prefer?

I think it would be nice to have a customizable option or configuration. :)

Or turn it on or off completely when the word is inserted.

**Comment by nyngwang** [comment]
@AGou-ops I think I just did. I myself love this feature too, and, I think this has to be the default, and no need to create a toggle (Yup I just changed my mind). I'm merging it.

Let me explain you more on this problem so you will realize that this is actually a bug (so why I prefer to not create a toggle, since we don't want to toggle on a bug, right?)

1. if you type `testing` on its own line, you will find that the entire word `testing` got highlighted when you're still in insert mode. And this is the expected beviour.
2. if you type `'testing'` on its own line, back to normal mode, and press `i` when your cursor is at the ending single quote, nothing got highlighted. This is what you have reported.
3. Now, the key to debug this is this try: if you typing `testing `, back to normal mode, and press `i` when your cursor is at the last space ` `, nothing got highlighted. This is the same as 2.

That conclusion is: when we're in insert mode, our cursor position is actually at the position(column) of the upcoming character. So it turns out this problem can be easily solved. To avoid performance decreasing on the working normal-mode highlighting (as the problem only happens in insert-mode), I separate the `autocmd` into two ones.

Let me know if anything is still wrong in the upcoming PR :)
